lib/workers/repository/process/index.ts

```diff
@@ -1,24 +1,67 @@
 import { mergeChildConfig } from '../../../config';
 import type { RenovateConfig } from '../../../config/types';
+import { CONFIG_VALIDATION } from '../../../constants/error-messages';
 import { logger } from '../../../logger';
 import type { PackageFile } from '../../../manager/types';
+import { platform } from '../../../platform';
+import { getCache } from '../../../util/cache/repository';
 import { branchExists } from '../../../util/git';
 import { addSplit } from '../../../util/split';
 import type { BranchConfig } from '../../types';
 import { readDashboardBody } from '../dependency-dashboard';
 import { ExtractResult, extract, lookup, update } from './extract-update';
 import type { WriteUpdateResult } from './write';
 
-function getBaseBranchConfig(
+async function getBaseBranchConfig(
   baseBranch: string,
   config: RenovateConfig
-): RenovateConfig {
+): Promise<RenovateConfig> {
   logger.debug(`baseBranch: ${baseBranch}`);
-  const baseBranchConfig = mergeChildConfig(config, { baseBranch });
-  if (config.baseBranches.length > 1) {
+
+  let baseBranchConfig: RenovateConfig = config;
+
+  if (
+    config.useBaseBranchConfig === 'replace' &&
+    baseBranch !== config.defaultBranch
+  ) {
+    logger.debug(
+      { baseBranch },
+      `Using config from base branch because useBaseBranchConfig option specified`
+    );
+
+    // Retrieve config file name autodetected for this repo
+    const cache = getCache();
+    const configFileName = cache.configFileName;
+
+    try {
+      baseBranchConfig = await platform.getJsonFile(
+        configFileName,
+        config.repository,
+        baseBranch
+      );
+    } catch (err) {
+      logger.error(
+        { configFileName, baseBranch },
+        `Error fetching config file from base branch - possible config name mismatch between branches?`
+      );
+
+      const error = new Error(CONFIG_VALIDATION);
+      error.validationSource = 'config';
+      error.validationError = 'Error fetching config file';
+      error.validationMessage = `Error fetching config file ${configFileName} from branch ${baseBranch}`;
+      throw error;
+    }
+
+    baseBranchConfig = mergeChildConfig(config, baseBranchConfig);
+    baseBranchConfig.baseBranches = config.baseBranches;
+  }
+
+  baseBranchConfig = mergeChildConfig(baseBranchConfig, { baseBranch });
+  if (baseBranchConfig.baseBranches.length > 1) {
```

Comment:
Are you sure about this? In theory `baseBranches` should be completely ignored in all but the default branch

Reply:
I thought that maybe branches would overlap, but now I think it should work without this block